IBAnimatable/SlideAnimator.swift

```diff
@@ -0,0 +1,98 @@
+    self.fromDirection = fromDirection
+    self.transitionDuration = transitionDuration
+    fade = params.contains("fade")
+    horizontal = fromDirection.isHorizontal
+    
+    switch fromDirection {
+    case .Right:
+      self.transitionAnimationType = .Slide(direction: .Right, params: params)
+      self.reverseAnimationType = .Slide(direction: .Left, params: params)
+      self.interactiveGestureType = .Pan(direction: .Left)
+      reverse = true
+    case .Top:
+      self.transitionAnimationType = .Slide(direction: .Top, params: params)
+      self.reverseAnimationType = .Slide(direction: .Bottom, params: params)
+      self.interactiveGestureType = .Pan(direction: .Bottom)
+      reverse = false
+    case .Bottom:
+      self.transitionAnimationType = .Slide(direction: .Bottom, params: params)
+      self.reverseAnimationType = .Slide(direction: .Top, params: params)
+      self.interactiveGestureType = .Pan(direction: .Top)
+      reverse = true
+    default:
+      self.transitionAnimationType = .Slide(direction: .Left, params: params)
+      self.reverseAnimationType = .Slide(direction: .Right, params: params)
+      self.interactiveGestureType = .Pan(direction: .Right)
+      reverse = false
+    }
+    super.init()
+  }
+}
+
+extension SlideAnimator: UIViewControllerAnimatedTransitioning {
+  public func transitionDuration(transitionContext: UIViewControllerContextTransitioning?) -> NSTimeInterval {
+    return retrieveTransitionDuration(transitionContext)
+  }
+  
+  public func animateTransition(transitionContext: UIViewControllerContextTransitioning) {
+    let (tempfromView, tempToView, tempContainerView) = retrieveViews(transitionContext)
+    guard let fromView = tempfromView, toView = tempToView, containerView = tempContainerView else {
+      transitionContext.completeTransition(true)
+      return
+    }
+    
+    
+    let travelDistance = horizontal ? containerView.bounds.width : containerView.bounds.height
+    let travel = CGAffineTransformMakeTranslation(horizontal ? (reverse ? travelDistance : -travelDistance) : 0, horizontal ? 0 : (reverse ? travelDistance : -travelDistance))
+    containerView.addSubview(toView)
+    if fade {
+      toView.alpha = 0
+    }
+    toView.transform = CGAffineTransformInvert(travel)
+    animateSlideTransition(toView: toView, fromView: fromView, travel: travel) {
+      fromView.transform = CGAffineTransformIdentity
+      transitionContext.completeTransition(!transitionContext.transitionWasCancelled())
+    }
+  }
+  
+}
+
+// MARK: - Animation
+
+private extension SlideAnimator {
+  
+  func animateSlideTransition(toView toView: UIView, fromView: UIView, travel: CGAffineTransform, completion: AnimatableCompletion) {
+    UIView.animateWithDuration(transitionDuration, animations: {
+      fromView.transform = travel
+      toView.transform = CGAffineTransformIdentity
+      if self.fade {
+        fromView.alpha = 0
+        toView.alpha = 1
+      }
+    }) { _ in
```

Comment:
I'm not used about this syntax (I'm generally following apple / github code style) but it's your call 👍